### ISSUE TITLE
ArrayIndexOutOfBoundsException for valid input

### ISSUE THREAD

**Issue by haozhun** [jruby/joni#28]
JONI fails with ArrayIndexOutOfBoundsException for pattern `^show\s*(\b.+\b)\s*vs\s*(\b.+\b)$` and input `show c`.

```
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: 6
	at io.airlift.jcodings.specific.UTF8Encoding.length(UTF8Encoding.java:35)
	at io.airlift.jcodings.specific.BaseUTF8Encoding.mbcToCode(BaseUTF8Encoding.java:91)
	at io.airlift.jcodings.specific.UTF8Encoding.mbcToCode(UTF8Encoding.java:24)
	at io.airlift.jcodings.Encoding.isMbcWord(Encoding.java:469)
	at io.airlift.joni.ByteCodeMachine.opWordBound(ByteCodeMachine.java:1063)
	at io.airlift.joni.ByteCodeMachine.matchAt(ByteCodeMachine.java:239)
	at io.airlift.joni.Matcher.matchCheck(Matcher.java:304)
	at io.airlift.joni.Matcher.searchInterruptible(Matcher.java:457)
	at io.airlift.joni.Matcher.search(Matcher.java:318)
```

Reproduction:
```
        byte[] pattern = "^show\\s*(\\b.+\\b)\\s*vs\\s*(\\b.+\\b)$".getBytes(StandardCharsets.UTF_8);
        byte[] str = ("show c").getBytes(StandardCharsets.UTF_8);
        Regex regex = new Regex(pattern, 0, pattern.length, Option.NEGATE_SINGLELINE, UTF8Encoding.INSTANCE, Syntax.Java);
        Matcher matcher = regex.matcher(str);
        int result = matcher.search(0, str.length, Option.DEFAULT);
        System.out.println(result);
```

**Comment by lopex** [comment]
Can't reproduce, can you confirm it works against latest joni release ?

**Comment by haozhun** [comment]
This was fixed in joni 2.1.7.
